src/test/regress/sql/multi_extension.sql

```diff
@@ -262,15 +262,52 @@ BEGIN
 END;
 $$;
 
+-- see that the deamon started
 SELECT datname,
     datname = current_database(),
     usename = (SELECT extowner::regrole::text FROM pg_extension WHERE extname = 'citus')
 FROM test.maintenance_worker();
 
--- Test that database with active worker can be dropped. That'll
--- require killing the maintenance worker.
+-- Test that database with active worker can be dropped.
 \c regression
-SELECT datname,
-    pg_terminate_backend(pid)
-FROM test.maintenance_worker('another');
+
+CREATE SCHEMA test_deamon;
+
+-- we create a similar function on the regression database
+-- note that this function checks for the existence of the daemon
+-- when not found, returns true else tries for 5 times and 
+-- returns false
+CREATE OR REPLACE FUNCTION test_deamon.maintenance_deamon_died(p_dbname text)
+    RETURNS boolean
+    LANGUAGE plpgsql
+AS $$
+DECLARE
+   activity record;
+   tryCount int := 0;
+BEGIN
+    LOOP
+        SELECT * INTO activity FROM pg_stat_activity
+        WHERE application_name = 'Citus Maintenance Daemon' AND datname = p_dbname;
+        IF activity.pid IS NULL THEN
+            RETURN true;
+        ELSE
+            PERFORM pg_sleep(0.1);
+            PERFORM pg_stat_clear_snapshot();
+        END IF;
+        tryCount = tryCount + 1;
+        IF tryCount > 5 THEN
+            RETURN false;
+        END IF;
```

Comment:
Do we have a bug that allows that? 

It seems like we can only have two cases:
1) `StopMaintenanceDaemon` runs before the worker acquires the lwlock. The worker sees its not in the hash and calls `proc_exit(0)`.
2) `StopMaintenanceDaemon` runs after the worker acquires the lwlock. The signals and `workerPid` set up and when StopMaintenanceDaemon calls `kill`, the worker calls `die`.